pkg/network/ebpf/c/tracer-stats.h

```diff
@@ -45,25 +46,52 @@ static __always_inline protocol_t get_protocol(conn_tuple_t *t) {
     conn_tuple_copy.netns = 0;
     conn_tuple_copy.pid = 0;
     protocol_t *cached_protocol_ptr = bpf_map_lookup_elem(&connection_protocol, &conn_tuple_copy);
+    if (cached_protocol_ptr != NULL) {
+       return *cached_protocol_ptr;
+    }
+
+    conn_tuple_t *cached_skb_conn_tup_ptr = bpf_map_lookup_elem(&conn_tuple_to_socket_skb_conn_tuple, &conn_tuple_copy);
```

Comment:
I'm having a hard-time understanding the purpose of this map. Are key and value equal except by namespace and pid? If that's the case it would make much more sense IMO to have a small helper like the following:

```
static __always_inline protocol_t get_protocol(conn_tuple_t *t) {
  ns = tup->ns;
  pid = tup->pid;
  tup->ns = 0;
  tup->pid = 0;
  cached_protocol_ptr = bpf_map_lookup_elem(&connection_protocol, &skb_tup);
  tup->ns = ns;
  tup->pid = pid;
  return cached_protocol_ptr
}
```

Reply:
when we are dealing with NATs we have different source (or dest) ip/port between the socket filter and the kprobes used by NPM
the map above is mapping between the connection tuple (without pid and ns, as those fields are not accessible in the context of socket filters) used by the kprobes (npm) to the connection tuple we have in the socket filter.

there is an extensive documentation in `protocol-classification-maps.h` above the map declaration, is it missing anything? can I make it clearer?

Reply:
my bad, I should have checked the map definition for the comments
we can definitely move this conversation to another channel since I think it is outside of the scope of this PR, but it sounds like that what you're doing in with this eBPF map + `tracepoint/net/net_dev_queue` is sort of the same what eBPF conntracker is doing, so I definitely think that we should somehow standardize this and use one or the other